src/components/Firestore/Requests/RequestDetails/index.tsx

```diff
@@ -14,19 +14,79 @@
  * limitations under the License.
  */
 
+import './index.scss';
+
 import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+import { AppState } from '../../../../store';
+import { getSelectedRequestEvaluationById } from '../../../../store/firestore/requests/evaluations/selectors';
+import { RequestDetailsRouteParams } from '../index';
+import { useRequestDetailedData, useRequestMainData } from '../utils';
+import RequestDetailsCodeViewer from './CodeViewer';
+import RequestDetailsHeader from './Header';
+import RequestDetailsInspectionSection from './InspectionSection';
+
+const mapStateToProps = (
+  state: AppState,
+  { requestId }: RequestDetailsRouteParams
+) => {
+  return {
+    selectedRequest: getSelectedRequestEvaluationById(state, requestId || ''),
+  };
+};
+interface PropsFromStore extends ReturnType<typeof mapStateToProps> {}
+interface Props extends PropsFromStore, RequestDetailsRouteParams {}
+
+export const RequestDetails: React.FC<Props> = ({
+  selectedRequest,
+  requestId,
+}) => {
+  const [
+    requestTimeComplete,
+    requestTimeFormatted,
+    requestMethod,
+    resourcePath,
+    outcomeData,
+  ] = useRequestMainData(selectedRequest);
+  const [
+    firestoreRules,
+    linesOutcome,
+    linesIssues,
+    inspectionElements,
+  ] = useRequestDetailedData(selectedRequest);
```

Comment:
I don't think we need a function that returns a tuple here. `firestoreRules = selectedRequest?.data?.rules;` is easier to read IMO. `inspectionElement` can probably use its own helper function (which can live in the same file).